src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useRef } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import useAnimatedRender from '../../hooks/useAnimatedRender'
+import { noop } from '../../utilities/other'
+import { manageTrappedFocus } from '../../utilities/focus'
+import {
+  BodyUI,
+  ClosePanelButtonUI,
+  FooterUI,
+  HeaderUI,
+  OverlayUI,
+  SidePanelUI,
+} from './SidePanel.css'
+import Button from '../Button/'
+import Icon from '../Icon'
+
+function SidePanel({
+  children,
+  className,
+  focusPanelOnShow = true,
+  mainActionButtonText = 'Start',
+  onMainActionClick = noop,
+  mainActionDisabled = false,
+  onClose = noop,
+  panelHeading = 'Review and Start',
+  panelSubHeading = 'Complete the required details before going live',
+  panelWidth = '400px',
+  show = false,
+  side = 'right',
+  trapFocus = false,
+  withHeader = true,
+  withFooter = true,
+  withOverlay = true,
+  zIndex = 999,
+}) {
+  const panelRef = useRef(null)
+  const overlayRef = useRef(null)
+  const [shouldRender, onAnimationEnd] = useAnimatedRender(
+    show,
+    overlayRef,
+    focusPanelOnShow && panelRef
+  )
+
+  function handleOverlayKeyDown(e) {
+    if (shouldRender && e.key === 'Escape') {
+      e.stopPropagation()
+      onClose()
+    } else if (e.key === 'Tab' && trapFocus && panelRef.current) {
+      manageTrappedFocus(panelRef.current, e)
+    }
+  }
+
+  return shouldRender ? (
+    <OverlayUI
+      className={classNames(
+        'SidePanel-Overlay',
+        show && 'element-in',
+        !withOverlay && 'no-overlay',
+        side,
+        className
+      )}
+      data-cy="sidepanel-overlay"
+      data-testid="sidepanel-overlay"
+      onAnimationEnd={onAnimationEnd}
+      onKeyDown={handleOverlayKeyDown}
+      ref={overlayRef}
+      zIndex={zIndex}
+    >
+      <SidePanelUI
+        aria-modal="true"
+        aria-labelledby="sidepanel-header-heading"
+        className="SidePanel"
+        data-cy="sidepanel"
+        data-testid="sidepanel"
+        id="sidepanel"
+        panelWidth={panelWidth}
+        ref={panelRef}
+        onFocus={e => {
+          console.log('FOCUSED')
```

Comment:
Stray console.log here. Also, should `onFocus` come as a prop to this component?

Reply:
This one is leftover from debugging... removed!